src/main/java/duke/exception/DukeException.java

```diff
@@ -0,0 +1,28 @@
+package duke.exception;
+
+/**
+ * Signals that an error has occurred during the usage of duke.
+ * This abstract class is the general class of exception
+ * produced by failed or interrupted operations.
+ *
+ * @author Zhi Bin
+ * @version Duke Level 9
+ */
+public abstract class DukeException extends Exception {
+
+    /**
+     * Returns a formatted message. Used by child
+     * class exception to format their error message.
+     *
+     * @param msg The message to be formatted.
+     * @return The formatted error message.
+     */
+    protected String formatMessage(String msg) {
+        String HORIZONTAL_LINE = "____________________________________________________________";
```

Comment:
If everything has indentation, why is indentation a separate thing? Setting indentation separately assumes that some lines do not need indentation but this does not seem to be the case.